ecs-cli/modules/config/config.go

```diff
@@ -74,43 +85,34 @@ func (cfg *CliConfig) ToServiceConfig() (*aws.Config, error) {
 
 // getCredentialProviders gets the chain of credentail provides to use when creating service clients.
 func (cfg *CliConfig) getCredentialProviders() []credentials.Provider {
-	credentialProviders := []credentials.Provider{}
-
-	// Fields saved in ecs-cli's config take precedence in the chain.
-	if cfg.AwsProfile != "" {
-		// ecs-cli has been configured to use an aws profile. Add that next.
-		credentialProviders = append(credentialProviders, &credentials.SharedCredentialsProvider{
-			Filename: "",
-			Profile:  cfg.AwsProfile,
-		})
-	} else {
-		// The 'profile' config is not set. Add static credential provider with
-		// credentials read from the config.
-		credentialProviders = append(credentialProviders, &credentials.StaticProvider{
+	// Append providers in the default credential providers chain to the chain.
+	// Order of credential resolution
+	// 1) Environment Variable provider
+	// 2) ECS Profile provider - attempts to fetch the credentials from the ECS config file
+	// 3) AWS profile - attempts to use the AWS profile specified in the ECS config file;
+	// If the AWS profile has not been specified, provider will attempt to use the 'default' profile
+	// 4) EC2 Instance role
+	credentialProviders := []credentials.Provider{
+		&credentials.EnvProvider{},
+		&credentials.StaticProvider{
 			Value: credentials.Value{
 				AccessKeyID:     cfg.AwsAccessKey,
 				SecretAccessKey: cfg.AwsSecretKey,
 			},
-		})
-	}
-
-	// Append providers in the default credential providers chain to the chain.
-	// EnvProvider is the first provider in the chain.
-	credentialProviders = append(credentialProviders, &credentials.EnvProvider{})
-
-	// The 'default' profile credential provider is next. Add if only if profile is not
-	// set in ecs config.
-	if cfg.AwsProfile == "" {
-		credentialProviders = append(credentialProviders, &credentials.SharedCredentialsProvider{
+		},
+		&credentials.SharedCredentialsProvider{
+			Filename: "",
+			Profile:  cfg.AwsProfile,
+		},
+		&credentials.SharedCredentialsProvider{
```

Comment:
I have removed the extra SharedCredentialsProvider such that the remaining SharedCredentialsProvider either attempts to use the specified Profile OR the default provider depending on whether the cfg.AwsProfile is non-empty.